prime-router/src/main/kotlin/history/azure/ReportFileFunction.kt

```diff
@@ -190,6 +205,83 @@ abstract class ReportFileFunction(
         }
     }
 
+    /**
+     * Function to return metadata of a single report from the CDC Intermediary.
+     * The [reportId] is a valid report UUID. This function is for the Intermediary only, please don't update
+     * without contacting that engineering team
+     */
+    fun retrieveETORIntermediaryMetadata(
+        request: HttpRequestMessage<String?>,
+        reportId: UUID,
+        context: ExecutionContext,
+        engine: HttpClientEngine?,
+    ): HttpResponseMessage {
+        val authResult = this.authSingleBlocks(request, reportId.toString())
+
+        if (authResult != null) {
+            return authResult
+        }
+
+        var response: HttpResponse?
+        val receiver = workflowEngine.settings.findReceiver(this.intermediaryReceiverName)
+        val client = if (engine == null) HttpClient() else HttpClient(engine)
+        val restTransportInfo = receiver?.transport as RESTTransportType
+        val (credential, jksCredential) = RESTTransport().getCredential(restTransportInfo, receiver)
+        val logger: Logger = context.logger
+        var authPair: Pair<Map<String, String>?, String?> =
+            Pair(null, null)
+
+        var responseBody = ""
+
+        runBlocking {
```

Comment:
Nit: I think the more kotlinnier way might be something like and avoid having the mutable state

```suggestion
         val authPair = runBlocking {
            async {
                authPair = RESTTransport().getOAuthToken(
                    restTransportInfo,
                    reportId.toString(),
                    jksCredential,
                    credential,
                    logger
                )
            }.await()
        }
```

Reply:
Fixed.